### DISCUSSION TITLE
Crazyflie 2.1 flight time and battery life variation with added decks

### DISCUSSION THREAD

**Discussion by MalshaDeZ** [bitcraze/discussions#1444] [Q&A] [answered]
Hi everyone!

 I'm planning to buy a crazyflie 2.1 drone for some research work. I'm trying to figure out the stats regarding the flight times. In the documentation it says "It can fly up to 7 minutes and depending on how you fly it, the time will decrease.", which I'm assuming is the case without any additional decks.

For my use case, I need to add the AI deck, and I'm trying to figure out how the flight time changes accordingly, considering a instance where I stream the video throughout the flight.

I tried the usual methods of searching for an answer. But there seems to be nothing concrete.

I believe someone must have done a proper analysis of how added weight changes the flight time.

If someone could give me a rough idea about the figures or point me in a direction where I could find the answers, I would really appreciate it.

thanks in advance

**Comment by tobbeanton** [comment, marked as answer]
First thing if you are flying with the AI-deck I would recommend the [thrust upgrade kit](https://store.bitcraze.io/collections/accessories/products/thrust-upgrade-bundle-for-crazyflie-2-x) and the [350mAh battery](https://store.bitcraze.io/collections/accessories/products/350mah-lipo-battery). This setup should get roughly 7 min of hover time.

> I believe someone must have done a proper analysis of how added weight changes the flight time.

From the [PWM to thurst](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/functional-areas/pwm-to-thrust/#observations) investigation one could probably approximativ that. However there are so many factors that influence the flight time so it will only be a rough estimate.

**Reply by MalshaDeZ** [reply]
@tobbeanton thank you very much!
